artiq/language/environment.py

```diff
@@ -69,7 +70,13 @@ def describe(self):
 
 class BooleanValue(_SimpleArgProcessor):
     """A boolean argument."""
-    pass
+    def process(self, x):
```

Comment:
Why not just `return bool(x)` ?

Reply:
Because the conversion rules for `bool()` are not intuitive. I don't know how many users know that `bool("abc")` or `bool(<any number except 0>)` evaluate to `True`. Conditions will still pass fine. I agree that it's heavy (is there a more compact way to write this?) but IMHO safer for the user.

Reply:
> I don't know how many users know that `bool("abc")` or `bool(<any number except 0>)` evaluate to `True`.

This is a well-documented Python feature and we can expect people reading this code to know about it.

Reply:
Also, I suspect that tests using ``is True/False`` only work due to a cpython implementation detail.

Reply:
> This is a well-documented Python feature and we can expect people reading this code to know about it.
Yes, but not necessarily to people writing experiments. What about going for the last suggestion and raising ValueError if `type(x) != bool`?

Reply:
Ok.

Reply:
The liberal and implicit casting in Python can be a bit confusing. I'm ok with the plan type check instead.

Reply:
Pushed. `TypeError` would make more sense from the python point of view but `ValueError` is what the other processors throw in that case.